service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/AddNoteAboutToSubmitControllerTest.java

```diff
@@ -0,0 +1,69 @@
+package uk.gov.hmcts.reform.fpl.controllers;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.boot.test.autoconfigure.OverrideAutoConfiguration;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.test.context.ActiveProfiles;
+import uk.gov.hmcts.reform.ccd.client.model.AboutToStartOrSubmitCallbackResponse;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+import uk.gov.hmcts.reform.fpl.model.CaseNote;
+import uk.gov.hmcts.reform.idam.client.IdamClient;
+import uk.gov.hmcts.reform.idam.client.models.UserDetails;
+
+import java.time.LocalDate;
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.when;
+import static uk.gov.hmcts.reform.fpl.utils.ElementUtils.unwrapElements;
+import static uk.gov.hmcts.reform.fpl.utils.ElementUtils.wrapElements;
+
+@ActiveProfiles("integration-test")
+@WebMvcTest(AddNoteController.class)
+@OverrideAutoConfiguration(enabled = true)
+class AddNoteAboutToSubmitControllerTest extends AbstractControllerTest {
+
+    @MockBean
+    IdamClient idamClient;
+
+    AddNoteAboutToSubmitControllerTest() {
+        super("add-note");
+    }
+
+    @BeforeEach
+    void setup() {
+        when(idamClient.getUserDetails(userAuthToken)).thenReturn(UserDetails.builder()
+            .forename("John")
+            .surname("Smith")
+            .build());
+    }
+
+    @Test
+    void shouldAddCaseNoteToList() {
+        CaseNote caseNote = caseNote(LocalDate.of(2019, 11, 12), "John Doe", "Existing note");
+
+        CaseDetails caseDetails = CaseDetails.builder()
+            .data(Map.of(
+                "caseNote", "Example case note",
+                "caseNotes", wrapElements(caseNote)))
+            .build();
+
+        AboutToStartOrSubmitCallbackResponse response = postAboutToSubmitEvent(caseDetails);
+        CaseData caseData = mapper.convertValue(response.getData(), CaseData.class);
+
+        assertThat(caseData.getCaseNote()).isNull();
+        assertThat(unwrapElements(caseData.getCaseNotes()))
+            .containsExactly(caseNote, caseNote(LocalDate.now(), "John Smith", "Example case note"));
```

Comment:
This might run into the rare occurrence of time based assertions being wrong, if the test runs at around midnight and the controller is called just before the day change but returns the result just after then the `LocalDate.now()` will have given different results when it is called in the service compared to when it is called here.

Reply:
I guess this is part of the bigger problem and I would maybe suggest about thinking about that separately...

Nightly builds run at  9:20:28 PM UTC.
You should be in tests not running tests at midnight 😛